### PR TITLE
Allow user to extend QueryBuilder

### PR DESCRIPTION
**Pull request by onomated**
*  `QueryBuilder.Extension` to support custom query builder modules
*  Added QueryBuilder extension macro test
*  Updated README

### CODE REVIEW

File: lib/query_builder/extension.ex [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  @moduledoc ~S"""
+  Use this module to create an extension module to `QueryBuilder` for app specific query utilities.
+  Use your query builder extension module wherever you would normally use `QueryBuilder`
+
+  Example:
+  ```
+  defmodule MyApp.QueryBuilder do
+    use QueryBuilder.Extension
+
+    defmacro __using__(opts) do
```
Comment (mathieuprog) [inline lib/query_builder/extension.ex:11]: We require the user to add that __using__ but actually we can inject that too, right?

Comment (onomated) [reply lib/query_builder/extension.ex:11]: The user technically doesn't need to make this a macro and `use MyApp.QueryBuilder`.  They can continue with `use QueryBuilder` in their models, and in the query functions utilize `MyApp.QueryBuilder.my_custom_function()`.  Also my macro does more than just `use QueryBuilder`.  A condensed version of my extension module:

```elixir
defmodule MyApp.QueryBuilder do
  use QueryBuilder.Extension
  require Ecto.Query
  import MyApp.QueryFunctions

  defmacro __using__(opts) do
    register_full_text_search_options(opts)

    quote do
      require QueryBuilder
      QueryBuilder.__using__(unquote(opts))
    end
  end

  # Add app specific query functions
  # ---------------------------------
  def search(query, search_term), do: do_search(query, search_term)
end
```

So I personally feel this shouldn't be as prescriptive as to what the user's macro functionality should be

Comment (onomated) [reply lib/query_builder/extension.ex:11]: @mathieuprog ^^^

**Comment by onomated** [comment]
@mathieuprog ping ^^^
I have commits to contribute that support offset and limit operations:  https://github.com/onomated/query_builder/commit/60131bdd74585d6610662ad2a9973608fac46844
https://github.com/onomated/query_builder/commit/0e2d3f3e1175e69908fd20e40e6734d5858d5f9c

File: README.md [hunk 2]
```diff
@@ -229,6 +248,57 @@ Blog.get_article_by_id(
 )
 ```
 
+## Extending Query Functions
+In the event that you want to extend QueryBuilder's functionality to include custom app specific query functions, there's the `QueryBuilder.Extension` macro to facilitate that.  You can create an extension module and use the `QueryBuilder.Extension` macro to inject all `QueryBuilder` functions into your custom module.  Any custom query functions added to your custom module are also available utilizing `QueryBuilder.from_list/2`.
```
Comment (mathieuprog) [inline README.md]: "the `QueryBuilder.Extension` macro" shouldn't that be "the `QueryBuilder.Extension` module"?

File: lib/query/limit.ex [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    apply_limit(ecto_query, value)
+  end
+
+  defp apply_limit(query, value) when is_integer(value) do
+    Ecto.Query.limit(query, ^value)
+  end
+
+  defp apply_limit(query, value) do
+    case Integer.parse(value) do
+      {int_value, __rest} ->
```
Comment (mathieuprog) [inline lib/query/limit.ex]: why the double underscore here? I think one underscore is more idiomatic in elixir for unused variables (just a small detail:))

Comment (onomated) [reply lib/query/limit.ex]: Yup.  Typo!

File: README.md (hunk 2 above)
Comment (onomated) [reply README.md]: I'll change to module.  The `use` is the macro.

**Comment by onomated** [comment]
@mathieuprog, Good to go.  PR also addresses #7.

**Pull request merged**